repository-hbase/src/main/scala/hbase/repository/HBaseAdminDataRepository.scala

```diff
@@ -0,0 +1,142 @@
+
+  @deprecated("Migrated to getAdminDataRest", "12 Dec 2017 - feature/HBase-Rest")
+  @throws(classOf[Exception])
+  private def getAdminData(referencePeriod: Option[YearMonth] = Some(HARDCODED_CURRENT_PERIOD), key: String): Option[AdminData] = {
+    val rowKey = RowKeyUtils.createRowKey(referencePeriod.getOrElse(HARDCODED_CURRENT_PERIOD), key)
+    Try(connector.getConnection.getTable(tableName)) match {
+      case Success(table: Table) =>
+        val scan = new Scan()
+          .setReversed(true)
+          .setRowPrefixFilter(Bytes.toBytes(key))
+        scan.setMaxResultSize(2)
+        table.get(new Get(Bytes.toBytes(rowKey))) match {
+          case res if res.isEmpty =>
+            logger.debug("No data found for row key '{}'", rowKey)
+            None
+          case result =>
+            logger.debug("Found data for row key '{}'", rowKey)
+            Some(convertToAdminData(result))
+        }
+      case Failure(e: Exception) =>
+        logger.error(s"Error getting data for row key $rowKey", e)
+        throw e
+    }
+  }
+
+  @throws(classOf[Exception])
+  private def getAdminDataRest(key: String, referencePeriod: Option[YearMonth], max: Long = maxResultSize): Future[PlayResult] = {
+    val headers = Seq("Accept" -> "application/json", "Authorization" -> s"Basic $auth")
+    val request = referencePeriod match {
+      case Some(r: YearMonth) =>
+        val rowKey = RowKeyUtils.createRowKey(r, key)
+        val uri = baseUrl / tableName.getNameWithNamespaceInclAsString / rowKey / columnFamily
+        ws.singleGETRequest(uri.toString, headers)
+      case None =>
+        val params = Seq("reversed" -> "true", "limit" -> max.toString)
+        val uri = baseUrl / tableName.getNameWithNamespaceInclAsString / key + "*"
+        ws.singleGETRequest(uri.toString, headers, params)
+    }
+    request.map {
+      case response if response.status == OK => {
+        val resp = (response.json \ "Row").as[Seq[JsValue]]
+        Try(resp.map(v => convertToAdminData(v))) match {
+          case Success(adminData: Seq[AdminData]) =>
+            //            Some(adminData)
+            Ok(Json.toJson(adminData)).as(JSON)
+          case Failure(ex) =>
+            // TODO - add exception type
+            //            None
+            BadRequest(errAsJson(BAD_REQUEST, "bad_request", s"$ex"))
+        }
+      }
+      case response if response.status == NOT_FOUND => NotFound(response.body).as(JSON)
+    }
+  }
+
+  private def convertToAdminData(result: JsValue): AdminData = {
+    val key = (result \ "key").as[String]
+    logger.debug(s"Found record $key")
+    val adminData: AdminData = RowKeyUtils.createAdminDataFromRowKey(decodeBase64(key))
+    val varMap = (result \ "Cell").as[Seq[JsValue]].map { cell =>
+      val column = decodeBase64((cell \ "column").as[String])
+      val value = decodeBase64((cell \ "$").as[String])
+      logger.debug(s"Found data column $column with value $value")
```

Comment:
Remove debug logging values of columns as IA say we can log VAT numbeers, PAYE references etc.